--- src/Nitrate_FormatWithEditorConfig/Terraria/ModLoader/UI/Elements/UIGrid.cs
+++ src/Nitrate/Terraria/ModLoader/UI/Elements/UIGrid.cs
@@ -190,7 +_,7 @@
 
 	public override List<SnapPoint> GetSnapPoints()
 	{
-		List<SnapPoint> list = [.. base.GetSnapPoint(out var item) ? [item] : []];
+		List<SnapPoint> list = [.. base.GetSnapPoint(out var item) ? new[] { item, } : [],];
 		foreach (var current in _items)
 		{
 			list.AddRange(current.GetSnapPoints());
