--- src/tModLoader/ReLogic/Peripherals/RGB/Logitech/LogitechDeviceGroup.cs
+++ src/Nitrate/ReLogic/Peripherals/RGB/Logitech/LogitechDeviceGroup.cs
@@ -9,9 +_,7 @@
 	private readonly List<RgbDevice> _devices = new List<RgbDevice>();
 	private bool _isInitialized;
 	private readonly VendorColorProfile _colorProfiles;
-#if NETCORE
 	private bool _initializationFailed;
-#endif
 
 	public LogitechDeviceGroup(VendorColorProfile colorProfiles)
 	{
@@ -23,7 +_,6 @@
 		if (_isInitialized)
 			return;
 
-#if NETCORE
 		if (_initializationFailed)
 			return;
 
@@ -36,7 +_,6 @@
 
 			return;
 		}
-#endif
 
 		try {
 			if (!NativeMethods.LogiLedInit())
