--- src/Nitrate_FormatWithEditorConfig/Terraria/Lang.cs
+++ src/Nitrate/Terraria/Lang.cs
@@ -1162,7 +_,8 @@
 	public static string LocalizedDuration(TimeSpan time, bool abbreviated, bool showAllAvailableUnits)
 	{
 		var text = "";
-		abbreviated |= !GameCulture.FromCultureName(GameCulture.CultureName.English).IsActive;
+		// TODO(NITRATE): Attach this behavior to GameCulture.
+		abbreviated |= !EN_US.IsActive;
 		if (time.Days > 0)
 		{
 			var num = time.Days;
