--- src/tModLoader/Terraria/Lang.cs
+++ src/Nitrate/Terraria/Lang.cs
@@ -9,6 +_,7 @@
 using Terraria.Localization;
 using Terraria.Map;
 using Terraria.ModLoader;
+using Terraria.Nitrate.Localization;
 using Terraria.UI;
 using Terraria.Utilities;
 
@@ -1076,7 +_,7 @@
 	public static string LocalizedDuration(TimeSpan time, bool abbreviated, bool showAllAvailableUnits)
 	{
 		string text = "";
-		abbreviated |= !GameCulture.FromCultureName(GameCulture.CultureName.English).IsActive;
+		abbreviated |= !Languages.en_US.IsActive;
 		if (time.Days > 0) {
 			int num = time.Days;
 			if (!showAllAvailableUnits && time > TimeSpan.FromDays(1.0))
