--- src/Nitrate_FormatWithEditorConfig/Terraria/NPC.cs
+++ src/Nitrate/Terraria/NPC.cs
@@ -39857,7 +_,7 @@
 
 	private void SpawnStardustMark_StardustTower()
 	{
-		List<int> list = [.. CountNPCS(405) + CountNPCS(406) < 2 ? [405] : [], .. CountNPCS(402) < 2 ? [402] : [], .. CountNPCS(407) < 1 ? [407] : []];
+		List<int> list = [.. CountNPCS(405) + CountNPCS(406) < 2 ? new[] { 405, } : [], .. CountNPCS(402) < 2 ? new[] { 402, } : [], .. CountNPCS(407) < 1 ? new[] { 407, } : []];
 
 		if (list.Count > 0)
 		{
@@ -39941,7 +_,7 @@
 
 	private void SpawnStardustMark_StardustWorm()
 	{
-		List<int> list = [.. CountNPCS(405) + CountNPCS(406) < 2 ? [405] : [], .. CountNPCS(402) < 3 ? [402] : [], .. CountNPCS(407) < 1 ? [407] : []];
+		List<int> list = [.. CountNPCS(405) + CountNPCS(406) < 2 ? new[] { 405, } : [], .. CountNPCS(402) < 3 ? new[] { 402, } : [], .. CountNPCS(407) < 1 ? new[] { 407, } : []];
 
 		if (list.Count <= 0)
 		{
