--- src/staging/tModLoader/Terraria/Terraria.csproj
+++ src/staging/Nitrate/Terraria/Terraria.csproj
@@ -68,6 +_,7 @@
 	<!-- Embedded Resources -->
 	<ItemGroup>
 		<EmbeddedResource Include="**\*.png" />
+		<EmbeddedResource Include="**\*.hjson" />
 		<EmbeddedResource Include="GameContent/Creative/Content/*" />
 		<EmbeddedResource Include="GameContent/Metadata/MaterialData/*" />
 		<EmbeddedResource Include="GameContent/WorldBuilding/*" />
