--- src/Nitrate_FormatWithEditorConfig/Terraria/GameContent/UI/States/UIEmotesMenu.cs
+++ src/Nitrate/Terraria/GameContent/UI/States/UIEmotesMenu.cs
@@ -273,10 +_,10 @@
 				70
 			},
 			72,
-			.. NPC.downedGoblins ? [64] : [],
-			.. NPC.downedFrost ? [66] : [],
-			.. NPC.downedPirates ? [65] : [],
-			.. NPC.downedMartians ? [71] : [],
+			.. NPC.downedGoblins ? new[] { 64, } : [],
+			.. NPC.downedFrost ? new[] { 66, } : [],
+			.. NPC.downedPirates ? new[] { 65, } : [],
+			.. NPC.downedMartians ? new[] { 71, } : [],
 		];
 
 		list.AddEmotesToCategory(EmoteID.Category.CrittersAndMonsters);
@@ -285,7 +_,7 @@
 
 	public static List<int> GetEmotesBosses()
 	{
-		List<int> list = [.. NPC.downedBoss1 ? [39] : []];
+		List<int> list = [.. NPC.downedBoss1 ? new[] { 39, } : [],];
 
 		if (NPC.downedBoss2)
 		{
