--- src/Nitrate_FormatWithEditorConfig/Terraria/_TML/Item.cs
+++ src/Nitrate/Terraria/_TML/Item.cs
@@ -190,7 +_,7 @@
 	public ref partial struct DisableCloneMethod
 	{
 		public DisableCloneMethod(string msg) => cloningDisabled = msg;
-		public static void Dispose() => cloningDisabled = null;
+		public readonly void Dispose() => cloningDisabled = null;
 	}
 
 	[ThreadStatic]
@@ -199,7 +_,7 @@
 	internal ref partial struct DisableNewItemMethod
 	{
 		internal DisableNewItemMethod(bool disabled) => newItemDisabled = disabled;
-		internal static void Dispose() => newItemDisabled = false;
+		internal readonly void Dispose() => newItemDisabled = false;
 	}
 
 	/// <summary>
