--- src/tModLoader/Terraria/ModLoader/FrameworkVersion.cs
+++ src/Nitrate/Terraria/ModLoader/FrameworkVersion.cs
@@ -15,22 +_,12 @@
 
 public static class FrameworkVersion
 {
-#if NETCORE
 	public static readonly Framework Framework = Framework.NetCore;
-#else
-	public static readonly Framework Framework = Framework.NetFramework;
-#endif
 
 	public static readonly Version Version = GetVersion();
 
 	private static Version GetVersion()
 	{
-#if !NETCORE
-		const string subkey = @"SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full\";
-		using (RegistryKey ndpKey = RegistryKey.OpenBaseKey(RegistryHive.LocalMachine, RegistryView.Registry32).OpenSubKey(subkey))
-			if (ndpKey != null && ndpKey.GetValue("Release") is int releaseKey)
-				return CheckFor45PlusVersion(releaseKey);
-#endif
 		return Environment.Version;
 	}
 
