--- src/Nitrate_FormatWithEditorConfig/Terraria/UI/FancyErrorPrinter.cs
+++ src/Nitrate/Terraria/UI/FancyErrorPrinter.cs
@@ -80,7 +_,7 @@
 				"If you are using an Anti-virus, please make sure it does not block Terraria in any way.",
 				"Try making sure your `Documents/My Games/Terraria` folder is not set to 'read-only'.",
 				"Try verifying integrity of game files via Steam.",
-				.. filePath.ToLower().Contains("onedrive") ? ["Try updating OneDrive."] : [],
+				.. filePath.ToLower().Contains("onedrive") ? new[] { "Try updating OneDrive.", } : [],
 			];
 
 			stringBuilder.AppendLine();
@@ -118,7 +_,7 @@
 				"If you are using an Anti-virus, please make sure it does not block Terraria in any way.",
 				"Try making sure your `Documents/My Games/Terraria` folder is not set to 'read-only'.",
 				"Try verifying integrity of game files via Steam.",
-				.. folderPath.ToLower().Contains("onedrive") ? ["Try updating OneDrive."] : [],
+				.. folderPath.ToLower().Contains("onedrive") ? new[] { "Try updating OneDrive.", } : [],
 			];
 
 			stringBuilder.AppendLine();
