--- src/staging/tModLoader/Terraria/Utilities/PlatformUtilities.cs
+++ src/staging/Nitrate/Terraria/Utilities/PlatformUtilities.cs
@@ -1,17 +_,11 @@
 using System;
 using System.IO;
-#if FNA
-using System.Runtime.InteropServices;
 using Microsoft.Xna.Framework.Graphics;
-using SDL2;
-#endif
 
 namespace Terraria.Utilities;
 
 public static class PlatformUtilities
 {
-#if NETCORE
-	// Use FNA3D.WritePNGStream to avoid SDL_image dependency
 	public unsafe static void SavePng(Stream stream, int width, int height, int imgWidth, int imgHeight, byte[] data)
 	{
 		if (width * height * 4 > data.Length)
@@ -20,7 +_,8 @@
 		fixed (byte* ptr = data)
 			FNA3D.WritePNGStream(stream, width, height, imgWidth, imgHeight, (IntPtr)ptr);
 	}
-#elif FNA
+
+	/*
 	private unsafe static IntPtr INTERNAL_getScaledSurface(byte[] data, int srcW, int srcH, int dstW, int dstH)
 	{
 		IntPtr intPtr = SDL.SDL_CreateRGBSurface(0u, srcW, srcH, 32, 255u, 65280u, 16711680u, 4278190080u);
@@ -59,5 +_,5 @@
 	{
 		throw new NotSupportedException("Use Bitmap to save png images on windows");
 	}
-#endif
+	*/
 }
