--- src/tModLoader/ReLogic/Peripherals/RGB/Razer/RazerDeviceGroup.cs
+++ src/Nitrate/ReLogic/Peripherals/RGB/Razer/RazerDeviceGroup.cs
@@ -10,9 +_,7 @@
 	private readonly List<RgbDevice> _devices = new List<RgbDevice>();
 	private bool _isInitialized;
 	private readonly VendorColorProfile _colorProfiles;
-#if NETCORE
 	private bool _initializationFailed;
-#endif
 
 	public RazerDeviceGroup(VendorColorProfile colorProfiles)
 	{
@@ -24,7 +_,6 @@
 		if (_isInitialized)
 			return;
 
-#if NETCORE
 		if (_initializationFailed)
 			return;
 
@@ -35,7 +_,6 @@
 
 			return;
 		}
-#endif
 
 		try {
 			RzResult rzResult = NativeMethods.Init();
