--- src/tModLoader/Terraria/Localization/LocalizedText.TML.cs
+++ src/Nitrate/Terraria/Localization/LocalizedText.TML.cs
@@ -25,6 +_,9 @@
 	//   other
 	public static int CardinalPluralRule(GameCulture culture, int count)
 	{
+		return culture.CardinalPluralRule(count);
+
+		/*
 		int mod_i10 = count % 10;
 		int mod_i100 = count % 100;
 		static bool contains(int i, int a, int b) => i >= a && i <= b;
@@ -69,6 +_,7 @@
 				// other
 				return 0;
 		}
+		*/
 	}
 
 	public static readonly Regex PluralizationPatternRegex = new Regex(@"{\^(\d+):([^\r\n]+?)}", RegexOptions.Compiled); // "{0} {^0:item;items}"
