--- src/tModLoader/Terraria/GameContent/UI/Elements/UIWorkshopPublishWorldListItem.cs
+++ src/Nitrate/Terraria/GameContent/UI/Elements/UIWorkshopPublishWorldListItem.cs
@@ -4,6 +_,7 @@
 using Terraria.GameContent.UI.States;
 using Terraria.IO;
 using Terraria.Localization;
+using Terraria.Nitrate.Localization;
 using Terraria.UI;
 
 namespace Terraria.GameContent.UI.Elements;
@@ -121,7 +_,7 @@
 		Utils.DrawBorderString(spriteBatch, expertText, vector + new Vector2(x2, num4 + 3f), gameModeColor);
 		vector.X += num3 + 5f;
 		float num5 = num2;
-		if (!GameCulture.FromCultureName(GameCulture.CultureName.English).IsActive)
+		if (!Languages.en_US.IsActive)
 			num5 += 40f;
 
 		DrawPanel(spriteBatch, vector, num5, height);
