--- src/Nitrate_FormatWithEditorConfig/Terraria/ModLoader/Logging.ExceptionHandling.cs
+++ src/Nitrate/Terraria/ModLoader/Logging.ExceptionHandling.cs
@@ -19,7 +_,7 @@
 	public readonly ref partial struct QuietExceptionHandle
 	{
 		public QuietExceptionHandle() { quietExceptionCount++; }
-		public static void Dispose() => quietExceptionCount--;
+		public readonly void Dispose() => quietExceptionCount--;
 	}
 
 	[ThreadStatic]
