--- src/staging/tModLoader/Terraria/Initializers/ChromaInitializer.cs
+++ src/staging/Nitrate/Terraria/Initializers/ChromaInitializer.cs
@@ -508,10 +_,6 @@
 
 		LoadSpecialRulesForDevices();
 		AppDomain.CurrentDomain.ProcessExit += OnProcessExit;
-#if !NETCORE
-		if (Platform.IsWindows)
-			Application.ApplicationExit += OnProcessExit;
-#endif
 	}
 
 	private static void LoadSpecialRulesForDevices()
