--- src/tModLoader/Terraria/ModLoader/UI/UICommon.cs
+++ src/Nitrate/Terraria/ModLoader/UI/UICommon.cs
@@ -8,6 +_,7 @@
 using Terraria.GameContent.UI.Elements;
 using Terraria.ID;
 using Terraria.Localization;
+using Terraria.Nitrate.UI;
 using Terraria.UI;
 
 namespace Terraria.ModLoader.UI;
@@ -143,6 +_,8 @@
 	{
 		Asset<Texture2D> LoadEmbeddedTexture(string name)
 			=> ModLoader.ManifestAssets.Request<Texture2D>($"Terraria.ModLoader.{name}");
+		Asset<Texture2D> LoadEmbeddedNitrateTexture(string name)
+			=> ModLoader.ManifestAssets.Request<Texture2D>($"Terraria.Nitrate.{name}");
 
 		ButtonErrorTexture = LoadEmbeddedTexture("UI.ButtonError");
 		//ButtonConfigTexture = LoadEmbeddedTexture("Config.UI.ButtonConfig");
@@ -173,5 +_,9 @@
 		InfoDisplayPageArrowTexture = LoadEmbeddedTexture("UI.InfoDisplayPageArrow");
 		tModLoaderTitleLinkButtonsTexture = LoadEmbeddedTexture("UI.tModLoaderTitleLinkButtons");
 		CopyCodeButtonTexture = LoadEmbeddedTexture("UI.CopyCodeButton");
+
+		VersionBrandingRenderer.Vanilla.Icon = LoadEmbeddedNitrateTexture("UI.Icon_Vanilla");
+		VersionBrandingRenderer.Tml.Icon = LoadEmbeddedNitrateTexture("UI.Icon_TML");
+		VersionBrandingRenderer.Nitrate.Icon = LoadEmbeddedNitrateTexture("UI.Icon_Nitrate");
 	}
 }
