--- src/staging/tModLoader/Terraria/ModLoader/Engine/LogArchiver.cs
+++ src/staging/Nitrate/Terraria/ModLoader/Engine/LogArchiver.cs
@@ -17,12 +_,10 @@
 {
 	private const int MAX_LOGS = 20;
 
-#if NETCORE
 	// Register support for encodings not present on .NET Core (necessary to port from .NET Framework)
 	static LogArchiver() {
 		Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
 	}
-#endif
 
 	/// <summary>
 	/// Attempt archiving logs.
