--- src/Nitrate_FormatWithEditorConfig/Terraria/ModLoader/Engine/NativeLibraries.cs
+++ src/Nitrate/Terraria/ModLoader/Engine/NativeLibraries.cs
@@ -74,14 +_,14 @@
 
 	[LibraryImport("kernel32.dll", SetLastError = true)]
 	[return: MarshalAs(UnmanagedType.Bool)]
-	static partial bool SetDefaultDllDirectories(int directoryFlags);
+	private static partial bool SetDefaultDllDirectories(int directoryFlags);
 
 	[LibraryImport("kernel32.dll", SetLastError = true, StringMarshalling = StringMarshalling.Utf16)]
-	static partial void AddDllDirectory(string lpPathName);
+	private static partial void AddDllDirectory(string lpPathName);
 
 	[LibraryImport("kernel32.dll", SetLastError = true, StringMarshalling = StringMarshalling.Utf16)]
 	[return: MarshalAs(UnmanagedType.Bool)]
-	static partial bool SetDllDirectory(string lpPathName);
+	private static partial bool SetDllDirectory(string lpPathName);
 
 	private const int LOAD_LIBRARY_SEARCH_DEFAULT_DIRS = 0x00001000;
 }
