--- src/Nitrate_Staging/Terraria/WorldBuilding/Actions.cs
+++ src/Nitrate/Terraria/WorldBuilding/Actions.cs
@@ -102,7 +_,7 @@
 		public override bool Apply(Point origin, int x, int y, params object[] args) => UnitApply(origin, x, y, args);
 	}
 
-	public partial class Custom(CustomPerUnitAction perUnit) : GenAction
+	public partial class Custom(GenBase.CustomPerUnitAction perUnit) : GenAction
 	{
 		private readonly CustomPerUnitAction _perUnit = perUnit;
 
