--- src/Nitrate_FormatWithEditorConfig/Terraria/Localization/_TML/LocalizedText.cs
+++ src/Nitrate/Terraria/Localization/_TML/LocalizedText.cs
@@ -25,6 +_,10 @@
 	//   other
 	public static int CardinalPluralRule(GameCulture culture, int count)
 	{
+		// NITRATE: Define plural rules in the GameCulture instance.
+		return culture.CardinalPluralRule(count);
+
+		/*
 		var mod_i10 = count % 10;
 		var mod_i100 = count % 100;
 		static bool contains(int i, int a, int b) => i >= a && i <= b;
@@ -78,6 +_,7 @@
 			// other
 			return 0;
 		}
+		*/
 	}
 
 	public static readonly Regex PluralizationPatternRegex = MyRegex(); // "{0} {^0:item;items}"
