--- src/Nitrate_FormatWithEditorConfig/ReLogic/OS/Windows/_TML/NativeMethods.cs
+++ src/Nitrate/ReLogic/OS/Windows/_TML/NativeMethods.cs
@@ -6,11 +_,11 @@
 internal static partial class NativeMethods
 {
 	[LibraryImport("kernel32.dll")]
-	static partial IntPtr GetConsoleWindow();
+	private static partial IntPtr GetConsoleWindow();
 
 	[LibraryImport("user32.dll")]
 	[return: MarshalAs(UnmanagedType.Bool)]
-	static partial bool ShowWindow(IntPtr hWnd, int nCmdShow);
+	private static partial bool ShowWindow(IntPtr hWnd, int nCmdShow);
 
 	public static void HideConsole()
 	{
