--- src/staging/tModLoader/Terraria/GameContent/UI/Elements/UICharacterListItem.cs
+++ src/staging/Nitrate/Terraria/GameContent/UI/Elements/UICharacterListItem.cs
@@ -8,6 +_,7 @@
 using Terraria.IO;
 using Terraria.Localization;
 using Terraria.ModLoader;
+using Terraria.Nitrate.Localization;
 using Terraria.Social;
 using Terraria.UI;
 
@@ -330,7 +_,7 @@
 		vector.X += num2 + 5f;
 		Vector2 vector3 = vector;
 		float num3 = 140f;
-		if (GameCulture.FromCultureName(GameCulture.CultureName.Russian).IsActive)
+		if (Languages.ru_RU.IsActive)
 			num3 = 180f;
 
 		DrawPanel(spriteBatch, vector3, num3);
