--- src/Nitrate_FormatWithEditorConfig/Terraria/GameContent/UI/Elements/UIList.cs
+++ src/Nitrate/Terraria/GameContent/UI/Elements/UIList.cs
@@ -154,7 +_,7 @@
 
 	public override List<SnapPoint> GetSnapPoints()
 	{
-		List<SnapPoint> list = [.. GetSnapPoint(out var point) ? [point] : []];
+		List<SnapPoint> list = [.. GetSnapPoint(out var point) ? new[] { point, } : [],];
 
 		foreach (var item in _items)
 		{
