--- src/Nitrate_FormatWithEditorConfig/Terraria/GameContent/UI/Elements/UIWorkshopPublishWorldListItem.cs
+++ src/Nitrate/Terraria/GameContent/UI/Elements/UIWorkshopPublishWorldListItem.cs
@@ -128,7 +_,7 @@
 		Utils.DrawBorderString(spriteBatch, expertText, vector + new Vector2(x2, num4 + 3f), gameModeColor);
 		vector.X += num3 + 5f;
 		var num5 = num2;
-		if (!GameCulture.FromCultureName(GameCulture.CultureName.English).IsActive)
+		if (!EN_US.IsActive)
 		{
 			num5 += 40f;
 		}
