--- src/tModLoader/Terraria/GameContent/UI/Elements/UIWorldListItem.cs
+++ src/Nitrate/Terraria/GameContent/UI/Elements/UIWorldListItem.cs
@@ -12,6 +_,7 @@
 using Terraria.ModLoader.Config;
 using Terraria.Social;
 using Terraria.UI;
+using Terraria.Nitrate.Localization;
 
 namespace Terraria.GameContent.UI.Elements;
 
@@ -431,7 +_,7 @@
 		Utils.DrawBorderString(spriteBatch, expertText, vector + new Vector2(x2, 3f), gameModeColor);
 		vector.X += num2 + 5f;
 		float num3 = 150f;
-		if (!GameCulture.FromCultureName(GameCulture.CultureName.English).IsActive)
+		if (!Languages.en_US.IsActive)
 			num3 += 40f;
 
 		DrawPanel(spriteBatch, vector, num3);
@@ -442,7 +_,7 @@
 		vector.X += num3 + 5f;
 		float num4 = innerDimensions.X + innerDimensions.Width - vector.X;
 		DrawPanel(spriteBatch, vector, num4);
-		string arg = ((!GameCulture.FromCultureName(GameCulture.CultureName.English).IsActive) ? _data.CreationTime.ToShortDateString() : _data.CreationTime.ToString("d MMMM yyyy"));
+		string arg = ((!Languages.en_US.IsActive) ? _data.CreationTime.ToShortDateString() : _data.CreationTime.ToString("d MMMM yyyy"));
 		string textValue2 = Language.GetTextValue("UI.WorldCreatedFormat", arg);
 		float x5 = FontAssets.MouseText.Value.MeasureString(textValue2).X;
 		float x6 = num4 * 0.5f - x5 * 0.5f;
