--- src/Nitrate_FormatWithEditorConfig/Terraria/GameContent/UI/Elements/UIWorkshopImportWorldListItem.cs
+++ src/Nitrate/Terraria/GameContent/UI/Elements/UIWorkshopImportWorldListItem.cs
@@ -173,7 +_,7 @@
 		if (_data._worldSizeName != null)
 		{
 			var num3 = 150f;
-			if (!GameCulture.FromCultureName(GameCulture.CultureName.English).IsActive)
+			if (!EN_US.IsActive)
 			{
 				num3 += 40f;
 			}
@@ -188,7 +_,7 @@
 
 		var num4 = innerDimensions.X + innerDimensions.Width - vector.X;
 		DrawPanel(spriteBatch, vector, num4);
-		var arg = ((!GameCulture.FromCultureName(GameCulture.CultureName.English).IsActive) ? _data.CreationTime.ToShortDateString() : _data.CreationTime.ToString("d MMMM yyyy"));
+		var arg = Language.ActiveCulture.FormatDateTime(_data.CreationTime);
 		var textValue2 = Language.GetTextValue("UI.WorldCreatedFormat", arg);
 		var x5 = FontAssets.MouseText.Value.MeasureString(textValue2).X;
 		var x6 = num4 * 0.5f - x5 * 0.5f;
