--- src/Nitrate_FormatWithEditorConfig/Terraria/GameContent/UI/Elements/UIWorkshopImportWorldListItem.cs
+++ src/Nitrate/Terraria/GameContent/UI/Elements/UIWorkshopImportWorldListItem.cs
@@ -166,7 +_,7 @@
 		vector.X += num2 + 5f;
 		if (_data._worldSizeName != null) {
 			float num3 = 150f;
-			if (!GameCulture.FromCultureName(GameCulture.CultureName.English).IsActive)
+			if (!EN_US.IsActive)
 				num3 += 40f;
 
 			DrawPanel(spriteBatch, vector, num3);
@@ -179,7 +_,7 @@
 
 		float num4 = innerDimensions.X + innerDimensions.Width - vector.X;
 		DrawPanel(spriteBatch, vector, num4);
-		string arg = ((!GameCulture.FromCultureName(GameCulture.CultureName.English).IsActive) ? _data.CreationTime.ToShortDateString() : _data.CreationTime.ToString("d MMMM yyyy"));
+		string arg = Language.ActiveCulture.FormatDateTime(_data.CreationTime);
 		string textValue2 = Language.GetTextValue("UI.WorldCreatedFormat", arg);
 		float x5 = FontAssets.MouseText.Value.MeasureString(textValue2).X;
 		float x6 = num4 * 0.5f - x5 * 0.5f;
