--- src/tModLoader/Terraria/Audio/LegacyAudioSystem.cs
+++ src/Nitrate/Terraria/Audio/LegacyAudioSystem.cs
@@ -67,7 +_,6 @@
 			if (!contentSource.HasAsset(assetPath))
 				continue;
 
-#if !SERVER
 			string extension = contentSource.GetExtension(assetPath);
 			string assetPathWithExtension = assetPath + extension;
 
@@ -95,7 +_,6 @@
 				Main.IssueReporter.AddReport(textToShow);
 				Main.IssueReporterIndicator.AttemptLettingPlayerKnow();
 			}
-#endif
 		}
 
 		return audioTrack;
