--- src/Nitrate_FormatWithEditorConfig/ReLogic/OS/Windows/NativeMethods.cs
+++ src/Nitrate/ReLogic/OS/Windows/NativeMethods.cs
@@ -150,8 +_,8 @@
 	[LibraryImport("gdi32.dll")]
 	public static partial int GetDeviceCaps(IntPtr hdc, DeviceCap nIndex);
 
-	[LibraryImport("shell32.dll")]
+	[DllImport("shell32.dll")]
-	private static partial int SHFileOperation(ref SHFILEOPSTRUCT FileOp);
+	private static extern int SHFileOperation(ref SHFILEOPSTRUCT FileOp);
 
 	private static bool Send(string path, FileOperationFlags flags)
 	{
