--- src/Nitrate_Staging/Terraria/GameContent/UI/LegacyMultiplayerClosePlayersOverlay.cs
+++ src/Nitrate/Terraria/GameContent/UI/LegacyMultiplayerClosePlayersOverlay.cs
@@ -43,7 +_,7 @@
 		var screenPosition2 = Main.screenPosition;
 		for (var i = 0; i < 255; i++)
 		{
-			if (!player[i].active || myPlayer == i || player[i].dead || LocalPlayer.team <= 0 || LocalPlayer.team != player[i].team)
+			if (!player[i].active || myPlayer == i || player[i].dead || player[myPlayer].team <= 0 || player[myPlayer].team != player[i].team)
 			{
 				continue;
 			}
@@ -98,7 +_,7 @@
 			namePlatePos += vector2 / 2f;
 			namePlatePos *= 1f / uIScale;
 			namePlatePos -= vector2 / 2f;
-			if (LocalPlayer.gravDir == -1f)
+			if (player[myPlayer].gravDir == -1f)
 			{
 				namePlatePos.Y = (float)screenHeight - namePlatePos.Y;
 			}
@@ -108,8 +_,8 @@
 				var num9 = 20f;
 				var num10 = -27f;
 				num10 -= (vector2.X - 85f) / 2f;
-				num5 = player[i].Center.X - LocalPlayer.Center.X;
+				num5 = player[i].Center.X - player[myPlayer].Center.X;
-				num6 = player[i].Center.Y - LocalPlayer.Center.Y;
+				num6 = player[i].Center.Y - player[myPlayer].Center.Y;
 				var num11 = (float)Math.Sqrt(num5 * num5 + num6 * num6);
 				if (!(num11 > (float)num))
 				{
