--- src/Nitrate_Staging/Terraria/GameContent/UI/NewMultiplayerClosePlayersOverlay.cs
+++ src/Nitrate/Terraria/GameContent/UI/NewMultiplayerClosePlayersOverlay.cs
@@ -105,7 +_,7 @@
 		var mouseTextColor = Main.mouseTextColor;
 		var teamColor = Main.teamColor;
 		_ = Main.screenPosition;
-		var player2 = LocalPlayer;
+		var player2 = player[myPlayer];
 		var num2 = (float)(int)mouseTextColor / 255f;
 		if (player2.team == 0)
 		{
