--- src/tModLoader/Terraria/Localization/LanguageManager.cs
+++ src/Nitrate/Terraria/Localization/LanguageManager.cs
@@ -161,8 +_,9 @@
 				LoadLanguageFromFileTextJson(text2, canCreateCategories: true);
 			}
 			catch (Exception) {
-				if (Debugger.IsAttached)
-					Debugger.Break();
+				// Nitrate: This is allowed to fail.
+				// if (Debugger.IsAttached)
+				// 	Debugger.Break();
 
 				Console.WriteLine("Failed to load language file: " + text);
 				break;
