--- src/staging/tModLoader/Terraria/ModLoader/Core/LoaderUtils.cs
+++ src/staging/Nitrate/Terraria/ModLoader/Core/LoaderUtils.cs
@@ -83,7 +_,6 @@
 	/// <summary> Calls static constructors on the provided type and, optionally, its nested types. </summary>
 	public static void ResetStaticMembers(Type type, bool recursive = true)
 	{
-#if NETCORE
 		var typeInitializer = type.TypeInitializer;
 
 		if (typeInitializer != null) {
@@ -109,9 +_,6 @@
 			typeInitializer.Invoke(null, null);
 			field.SetValue(fieldObj, oldFlagValue);
 		}
-#else
-		type.TypeInitializer?.Invoke(null, null);
-#endif
 
 		if (recursive) {
 			foreach (var nestedType in type.GetNestedTypes()) {
